stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java

```diff
@@ -136,6 +147,56 @@ public static Object openReadableFile(String pathUrl) {
         return channel;
     }
 
+    public static Object getResource(String pathUrl) {
+        String[] className = Thread.currentThread().getStackTrace()[4].getClassName().split("\\.");
+        String resourcePath = RESOURCES + File.separator + className[0] + File.separator + className[1]
+                + File.separator + DEFAULT_BAL_PROJECT_VERSION + File.separator + pathUrl;
+        String executablePath = TARGET + File.separator + BIN + File.separator + className[1] + JAR;
+
+        InputStream resourceAsStream = ByteChannelUtils.class.getClassLoader().getResourceAsStream(resourcePath);
+        if (resourceAsStream == null) {
+            return IOUtils.createError("No such file : " + pathUrl);
+        }
+
+        try (ZipFile zipFile = new ZipFile(executablePath)) {
+            ZipEntry entry = zipFile.getEntry(resourcePath);
+            if (entry.isDirectory()) {
+                return IOUtils.createError("Found Directory : " + pathUrl + " , Required file.");
+            }
+            ReadableByteChannel readableByteChannel = Channels.newChannel(resourceAsStream);
+            Channel channel = new BlobIOChannel(new BlobChannel(readableByteChannel));
+            return createChannel(channel);
+        } catch (IOException e) {
+            return IOUtils.createError("Error unzipping file: " + executablePath);
+        }
+    }
+
+    public static ArrayValue getResources() {
+        ArrayList<String> list = new ArrayList<>();
```

Comment:
Can't we declare this as a List<String> type?

Reply:
Yes, we can declare this as List type and this is fixed.